.circleci/config.yml

```diff
@@ -0,0 +1,240 @@
+version: 2.1
+
+parameters:
+  ci_builder_image:
+    type: string
+    # depends with rvsol/lib/optimism submodule version
+    default: us-docker.pkg.dev/oplabs-tools-artifacts/images/ci-builder:v0.46.1
+
+workflows:
+  main:
+    jobs:
+      - pnpm-monorepo
+      - go-mod-download-monorepo
+      - go-mod-download-asterisc
+      - op-program-riscv:
+          requires: ["go-mod-download-monorepo"]
+      - asterisc-prestate:
+          requires: ["go-mod-download-asterisc", "op-program-riscv", "pnpm-monorepo"]
+
+jobs:
+  pnpm-monorepo:
+    docker:
+      - image: <<pipeline.parameters.ci_builder_image>>
+    resource_class: xlarge
+    steps:
+      - checkout
+      - run:
+          name: Fetch submodules for asterisc
+          command: git submodule update --init --recursive
+      - run:
+          name: Check L1 geth version
+          command: ./ops/scripts/geth-version-checker.sh || (echo "geth version is wrong, update ci-builder"; false)
+          working_directory: rvsol/lib/optimism
+      - run:
+          name: git submodules for monorepo
+          command: make submodules
+          working_directory: rvsol/lib/optimism
+      - restore_cache:
+          name: Restore PNPM Package Cache
+          keys:
+            - pnpm-packages-v2-{{ checksum "rvsol/lib/optimism/pnpm-lock.yaml" }}
+      - restore_cache:
+          name: Restore Go modules cache for monorepo
+          # this go mod cache wil be populated from go-mod-download-monorepo step
+          key: gomod-{{ checksum "rvsol/lib/optimism/go.sum" }}
+      # Fetch node_modules into the pnpm store
+      # This will cache node_modules based on pnpm-lock so other steps can instantly install them with `pnpm install --prefer-offline`
+      # --prefer-offline installs node_modules instantly by just reading from cache if it exists rather than fetching from network
+      # when installing node_modules pnpm simply adds symlinks instead of copying the files which is why it is pretty much instant to run --prefer-offline
+      # this allows a caching strategy of only checking pnpm-lockfile so we don't have to keep it in sync with our packages
+      # For more information see https://pnpm.io/cli/fetch
+      - run:
+          name: Fetch dependencies
+          command: pnpm fetch --frozen-lockfile --prefer-offline
+          working_directory: rvsol/lib/optimism
+      - save_cache:
+          name: Save PNPM Package Cache
+          key: pnpm-packages-v2-{{ checksum "rvsol/lib/optimism/pnpm-lock.yaml" }}
+          paths:
+            - "rvsol/lib/optimism/node_modules"
+      - run:
+          name: Install dependencies
+          command: pnpm install:ci:offline
+          working_directory: rvsol/lib/optimism
+      - run:
+          name: print forge version
+          command: forge --version
```

Comment:
Is this step just for logging?

Reply:
Yes. I think It is worth to check foundry version, and monorepo does it also.